tensordict/prototype/tensorclass.py

```diff
@@ -270,7 +269,8 @@ def wrapper(cls, tensordict, non_tensordict=None):
         # empty tensordict and writing values to it. we can skip this because we already
         # have a tensordict to use as the underlying tensordict
         tc = cls.__new__(cls)
-        tc.__dict__["_tensordict"] = tensordict
+        assert isinstance(td, TensorDictBase), td
```

Comment:
I think it's better to avoid `assert` statements for input validation because they can be disabled with certain interpreter flags. We can raise a `TypeError` or something instead.

Reply:
yeah it was for debugging thanks for spotting this